csrc/scheduler/normalization_utils.cpp

```diff
@@ -683,24 +683,29 @@ getOptionalInnerOuterPersistentBufferBatches(
   const int64_t threads_per_block_max = getThreadsPerSMGivenRegPerThread(255l);
   const int64_t batch_min = getMinimumBatch();
   const int64_t batch_max = getMaximumInnerOuterPersistentBufferBatch();
-
+  // increase batch_max by 1 to allow a small amount of register spills to keep
+  // a small threads_per_block so each SM can host more blocks. This can
+  // increase performance by 10% for edge cases, e.g. hidden size in the range
+  // of (7K,8K].
+  const int64_t batch_max_plus_one =
+      vectorize_factor > 1 ? batch_max + 1 : batch_max;
```

Comment:
Why `vectorize_factor > 1`? 

Shouldn't this adjustment be done inside `getMaximumInnerOuterPersitentBufferBatch`?

Reply:
`vectorize_factor > 1` is becuase we need to disable this optimization if vectorize_factor is 1 due to high register usage in cases can't be vectorized. added comments.

The function `getMaximumInnerOuterPersitentBufferBatch` is designed to determine the batch size without causing register spills. I think writing the adjustments outside of this function clarifies the intention to permit register spills. Moreover, the value returned by this function is further utilized within the same function to prevent segmentation by accommodating more register spills.

Reply:
Agree! These two changes are for different reasons. (1) The frist increase of 1 is to achieve higher occupancy. (2) The 2nd increase of 3 is to avoid segmentation. I changed these two paras to a more meaningfule name.